frontend/pages/hosts/details/cards/Software/helpers.tsx

```diff
@@ -194,55 +206,77 @@ export const getUiStatus = (
     return "failed_uninstall";
   }
 
-  // If installation is pending
+  // 3. Pending install/update
   if (status === "pending_install") {
     if (
       installed_versions &&
       installed_versions.length > 0 &&
       installerVersion
     ) {
-      // Are we updating (installerVersion > installed), or reinstalling (installerVersion == installed)?
       const isUpdate = installed_versions.some(
         (iv) => compareVersions(iv.version, installerVersion) === -1
       );
-
-      // Updating to a newer version
       if (isUpdate) {
         return isHostOnline ? "updating" : "pending_update";
       }
     }
-    // Reinstalling equivalent versions or installing with no currently installed versions
     return isHostOnline ? "installing" : "pending_install";
   }
 
-  // If uninstallation is pending
+  // 4. Pending uninstall
   if (status === "pending_uninstall") {
-    // Return 'uninstalling' if host is online, else 'pending_uninstall'
     return isHostOnline ? "uninstalling" : "pending_uninstall";
   }
 
-  // Check if any installed version is less than the installer version, indicating an update is available
+  // **Recently_uninstalled check comes BEFORE update_available**
+  if (software.status === null && lastUninstallDate && hostSoftwareUpdatedAt) {
+    const newerDate = getNewerDate(hostSoftwareUpdatedAt, lastUninstallDate);
+    if (newerDate === lastUninstallDate) {
+      return "recently_uninstalled";
+    }
+  }
+
+  // 5. Update available and recently updated
   if (
     installerVersion &&
     installed_versions &&
     installed_versions.some(
       (iv) => compareVersions(iv.version, installerVersion) === -1
     )
   ) {
-    return "update_available";
+    if (!lastInstallDate) {
+      return "update_available";
+    }
+    const newerDate =
+      hostSoftwareUpdatedAt && lastInstallDate
```

Comment:
```suggestion
      hostSoftwareUpdatedAt
```

Since L247 checks the inverse.